src/vos/vos_io.c

```diff
@@ -1759,6 +1760,13 @@ akey_update(struct vos_io_context *ioc, uint32_t pm_ver, daos_handle_t ak_toh,
 			return rc;
 		}
 		flags |= SUBTR_EVT;
+	} else if (ioc->ic_ec && iod->iod_recxs != NULL) {
```

Comment:
why do we put this akey in the list at all?  Is it to avoid refactoring the upper layer to remove an iod from the list ?

Would it be clearer if we set a iod_flag or some such?   At the very least, I think some more comments in the VOS code around the usage of gsize would be useful.

Reply:
I tested to remove/skip this akey in the iod list, but a problem is client-side bulk handle array are in original iod/akey order, if change like that, need another data structure to store which iod/akey is skipped before doing bulk transferring, that will need add some other trivial code.
Using "iod->iod_recxs" to pass singv gsize was already in current master code, this patch just add another special VOS_EC_SINGV_UPDATE_SKIP value for the skip case.
I'll add some comments, please check if it is fine with you, thanks.